H2MLauncher.Core/ViewModels/ServerBrowserViewModel.cs

```diff
@@ -74,20 +80,33 @@ public ServerBrowserViewModel(
             CheckUpdateStatusCommand = new AsyncRelayCommand(CheckUpdateStatusAsync);
             CopyToClipBoardCommand = new RelayCommand<ServerViewModel>(DoCopyToClipBoardCommand);
             SaveServersCommand = new AsyncRelayCommand(SaveServersAsync);
+            OpenUpdatePageInBrowserCommand = new AsyncRelayCommand(DoOpenUpdatePageInBrowserCommand, () => _updateStatus != "" && _updateStatus != "Restart for new version!");
             _logger = logger;
             _saveFileService = saveFileService;
             _errorHandlingService = errorHandlingService;
-            OpenUpdatePageInBrowserCommand = new RelayCommand(DoOpenUpdatePageInBrowserCommand, () => _updateStatus != "");
         }
 
-        private void DoOpenUpdatePageInBrowserCommand()
+        private async Task DoOpenUpdatePageInBrowserCommand()
         {
-            string destinationurl = "https://github.com/Bowhza/H2M-Launcher/releases/latest";
-            ProcessStartInfo sInfo = new(destinationurl)
+            ProgressBarVisibility = "Visible";
+            await _h2MLauncherService.UpdateLauncherToLatestVersion((double progress) =>
             {
-                UseShellExecute = true,
-            };
-            Process.Start(sInfo);
+                UpdateDownloadProgress = progress;
+                if (progress == 100)
+                {
+                    ProgressBarVisibility = "Hidden";
+                    UpdateStatus = "Restart for new version!";
+                }
+            }, CancellationToken.None).ConfigureAwait(false);
+
+            // NOTE: we could still show the release page with the newest features or in the APP?
```

Comment:
Good idea, but dont open in the app. Maybe show a dialog "Updated" with button "Restart"  and checkbox "Open release notes"